databases/week2/homework-part-3.sql

```diff
@@ -0,0 +1,36 @@
+USE hyf_lesson2;
+
+-- Get all the tasks assigned to users whose email ends in @spotify.com
+SELECT task.title, user.name, user.email
+FROM task
+JOIN user_task
+ON task.id = user_task.task_id
+JOIN user
+ON user_task.user_id = user.id
+WHERE user.email LIKE "%@spotify.com";
+	
+-- Get all the tasks for 'Donald Duck' with status 'Not started
+SELECT user.name, task.title, status.name
+FROM task
+JOIN user_task
+ON task.id = user_task.task_id
+JOIN status
+ON task.status_id = status.id
+JOIN user
+ON user_task.user_id = user.id
+WHERE user.name = "Donald Duck" AND status.name = "Done";
+
+-- Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+SELECT user.name, task.title, task.created
+FROM user_task
+JOIN task
+ON user_task.task_id = task.id
+JOIN user
+ON user_task.user_id = user.id
+WHERE user.name = "Maryrose Meadows" AND month(task.created) = 9 AND MONTH;
+
+-- Find how many tasks where created in each month, e.g. how many tasks were created in october, how many tasks were created in november, etc. (hint: use group by)
+SELECT monthname(created), count(*) AS number_of_tasks
+FROM task
+GROUP BY month(created)
+ORDER BY month(created);
```

Comment:
This one also gives me an error
```
Error Code: 1055. Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'hyf_lesson2.task.created' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
```
Can you figure out why it gives an error?

Reply:
I implemented all the feedback but this one.
In my pc I get exactly what i suppose to get with no error at all, so I am a bit confused with this one..

Reply:
We mentioned this in the general homework feedback during the last lesson. Not sure if we explained it well enough though :)

Basically the error says that you should group by the thing that you want to select. 
So in your query you would either do

```sql
SELECT month(created), count(*) AS number_of_tasks
FROM task
GROUP BY month(created)
ORDER BY month(created);
```

or

```sql
SELECT monthname(created), count(*) AS number_of_tasks
FROM task
GROUP BY monthname(created)
ORDER BY monthname(created);
```